Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/ProximityChecker.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Informs about the proximity staus of the local player
+    /// </summary>
+    public class ProximityChecker : MonoBehaviour
+    {
+        [SerializeField, Tooltip("At which distance will the player be considered 'close'?")]
+        float m_ActivationRadius = 1;
+
+        [SerializeField, Tooltip("A visual representation of the radius?")]
+        Transform m_RadiusRepresentation;
+        Transform m_Transform;
+
+        event Action<bool> OnLocalPlayerProximityStatusChanged;
+        internal bool LocalPlayerIsClose { get; private set; }
+        void Awake()
+        {
+            m_Transform = transform;
+            if (m_RadiusRepresentation)
+            {
+                const float k_OffsetFromGround = 0.01f;
+                m_RadiusRepresentation.transform.localPosition = new Vector3(0, (m_Transform.lossyScale.y / -2) + k_OffsetFromGround, 0);
+            }
+        }
+
+        internal void AddListener(Action<bool> callback)
+        {
+            OnLocalPlayerProximityStatusChanged += callback;
+        }
+
+        internal void RemoveListener(Action<bool> callback)
+        {
+            OnLocalPlayerProximityStatusChanged -= callback;
+        }
+
+        void Update()
+        {
+            if (m_RadiusRepresentation)
+            {
+                m_RadiusRepresentation.localScale = new Vector3(m_ActivationRadius * 2, m_RadiusRepresentation.localScale.y, m_ActivationRadius * 2);
+            }
+            bool oldValue = LocalPlayerIsClose;
+            LocalPlayerIsClose = LocalPlayerIsCloseEnough(m_Transform.position, m_ActivationRadius);
+            if (oldValue != LocalPlayerIsClose)
+            {
+                OnLocalPlayerProximityStatusChanged?.Invoke(LocalPlayerIsClose);
+            }
+        }
+
+        bool LocalPlayerIsCloseEnough(Vector3 point, float range)
+        {
+            if (!PlayerManager.s_LocalPlayer)
+            {
+                return false;
+            }
+            return Vector3.Distance(point, PlayerManager.s_LocalPlayer.transform.position) < range;
+        }
```

Comment:
I'm curious again, what is the Netcode-related learnings here with this use-case, as in, what friction are we aiming to solve here? For me, the use-case doesn't go much beyond adding a local player flag when performing a proximity check.  Just want to make sure I'm not missing any context.

Reply:
The learning is the answer to "how can I make thing X happen only when the local player is close enough?", which is a question directly and indirectly echoed by our community in forums and discords whenever they try to implement player-world interactions in a multiplayer environment.

Reply:
but that's not different from singleplayer though, that's not in our scope. We're not teaching people how to program, our target users are devs with a few years of xp trying to learn multiplayer.
There could maybe be some netcode specific lessons around client vs server positions not being the same? And having range checks not resulting in the same behaviour client and server side? We'd need to refactor this lesson for this though. Just the way it is, I don't see why this would be in a netcode sample.
Maybe we're missing something. Having comments in code here explaining the multiplayer lessons would be great, cause I don't see it right now.

Reply:
> I'm curious again, what is the Netcode-related learnings here with this use-case, as in, what friction are we aiming to solve here? 

The more specific, original netcode-specific, learning was "initializing a reference to the local player in `OnNetworkSpawn` ".

I now refactored this method to show how to use `NetworkManager.Singleton.LocalClient.PlayerObject` (another netcode-specific feature) instead of a custom flag, that should be enough to address your concerns.